File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const _counter = (nums) => {
            const counts = {};
            for (const num of nums) {
                if (num in counts) counts[num]++;
                else counts[num] = 1;
            }
            return counts;
        }
        const countEntries = Object.entries(_counter(nums)).sort((a,b) => a[1] - b[1]);
        let solution = [];
        for (let i = 0; i < k; i++) {
            solution.push(countEntries[countEntries.length - 1][0]);
            countEntries.pop();
        }
        return solution;
    }
}

File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        max_count = 0
        max_num = None
        freq = {}
        for num in nums:
            freq[num] = freq.get(num, 0) + 1
            if freq[num] >= len(nums) / 2:
                return num
            
            if freq[num] > max_count:
                max_num = num
                max_count = freq[num]
            
        return max_num
